src/java.base/share/classes/com/sun/crypto/provider/CipherCore.java

```diff
@@ -1223,4 +1284,55 @@ void updateAAD(byte[] src, int offset, int len) {
         checkReinit();
         cipher.updateAAD(src, offset, len);
     }
+
+    private static String privilegedGetProperty(final String property) {
+        return AccessController.doPrivileged(new PrivilegedAction<String>() {
+            public String run() {
+                return System.getProperty(property);
+            }
+        });
+    }
+
+    static {
+        String nativeCryptTrace = privilegedGetProperty("jdk.nativeCryptoTrace");
+        String nativeCryptStr   = privilegedGetProperty("jdk.nativeCrypto");
+        String nativeCBCStr     = privilegedGetProperty("jdk.nativeCBC");
+        String nativeGCMStr     = privilegedGetProperty("jdk.nativeGCM");
+
+        useNativeCrypto = Boolean.parseBoolean(nativeCryptStr) || (nativeCryptStr == null);
```

Comment:
This pattern should be fixed (here and elsewhere) to do the `null` check first.

Reply:
@keithc-ca I can make these changes but what is the reason that it has to be done this way. Reading on forums etc they all seem to say that the order of these is not really relevant in Java (as it was in C). For me it is more readable with the variable first. Please point me to where this is documented as being required for our code.

Reply:
OK, I think I may have misunderstood what you were saying here. 
Think you meant that the check for  
(nativeCryptStr == null)
should come before the check for
Boolean.parseBoolean(nativeCryptStr)

On my initial read I thought you were saying we needed to have 
(null == nativeCryptStr)
rather than 
(nativeCryptStr == null)

Will fix the code that was added (which was initially added by  someone else - so please do not presume all these issues are my coding, I am just porting other peoples code!!).

Reply:
This is probably the same as in Java 11 as that is where the code was copied from!!

Reply:
I meant the null check should be performed before the call to `parseBoolean()` as in
```
useNativeCrypto = (nativeCryptStr == null) || Boolean.parseBoolean(nativeCryptStr);
```
This would be more important if `parseBoolean(null)` threw an NPE.